src/cvode/cvode.c

```diff
@@ -2805,46 +2796,51 @@ static int cvNls(CVodeMem cv_mem, int nflag)
   }
 
   /* initial guess for the correction to the predictor */
-  N_VConst(ZERO, cv_mem->cv_acor);
+  SUNCheckCallLastErrNoRet(N_VConst(ZERO, cv_mem->cv_acor));
 
   /* call nonlinear solver setup if it exists */
   if ((cv_mem->NLS)->ops->setup) {
-    flag = SUNNonlinSolSetup(cv_mem->NLS, cv_mem->cv_acor, cv_mem);
-    if (flag < 0) return(CV_NLS_SETUP_FAIL);
-    if (flag > 0) return(SUN_NLS_CONV_RECVR);
+    nls_status = SUNNonlinSolSetup(cv_mem->NLS, cv_mem->cv_acor, cv_mem);
+    if (nls_status < 0) return(CV_NLS_SETUP_FAIL);
+    if (nls_status > 0) return(SUN_NLS_CONV_RECVR);
   }
 
   /* solve the nonlinear system */
-  flag = SUNNonlinSolSolve(cv_mem->NLS, cv_mem->cv_zn[0], cv_mem->cv_acor,
-                           cv_mem->cv_ewt, cv_mem->cv_tq[4], callSetup, cv_mem);
+  nls_status =
+    SUNCheckCallLastErrNoRet(SUNNonlinSolSolve(cv_mem->NLS, cv_mem->cv_zn[0],
+                                               cv_mem->cv_acor, cv_mem->cv_ewt,
+                                               cv_mem->cv_tq[4], callSetup,
+                                               cv_mem));
 
   /* increment counters */
-  (void) SUNNonlinSolGetNumIters(cv_mem->NLS, &nni_inc);
+  SUNCheckCallNoRet(SUNNonlinSolGetNumIters(cv_mem->NLS, &nni_inc));
   cv_mem->cv_nni += nni_inc;
 
-  (void) SUNNonlinSolGetNumConvFails(cv_mem->NLS, &nnf_inc);
+  SUNCheckCallNoRet(SUNNonlinSolGetNumConvFails(cv_mem->NLS, &nnf_inc));
   cv_mem->cv_nnf += nnf_inc;
 
   /* if the solve failed return */
-  if (flag != SUN_NLS_SUCCESS) return(flag);
+  if (nls_status != SUN_NLS_SUCCESS) return(nls_status);
 
   /* solve successful */
 
   /* update the state based on the final correction from the nonlinear solver */
-  N_VLinearSum(ONE, cv_mem->cv_zn[0], ONE, cv_mem->cv_acor, cv_mem->cv_y);
+  SUNCheckCallLastErrNoRet(N_VLinearSum(ONE, cv_mem->cv_zn[0], ONE, cv_mem->cv_acor, cv_mem->cv_y));
 
   /* compute acnrm if is was not already done by the nonlinear solver */
-  if (!cv_mem->cv_acnrmcur)
-    cv_mem->cv_acnrm = N_VWrmsNorm(cv_mem->cv_acor, cv_mem->cv_ewt);
+  if (!cv_mem->cv_acnrmcur) {
+    cv_mem->cv_acnrm = SUNCheckCallLastErrNoRet(N_VWrmsNorm(cv_mem->cv_acor, cv_mem->cv_ewt));
+  }
 
   /* update Jacobian status */
   cv_mem->cv_jcur = SUNFALSE;
 
   /* check inequality constraints */
-  if (cv_mem->cv_constraintsSet)
-    flag = cvCheckConstraints(cv_mem);
+  if (cv_mem->cv_constraintsSet) {
+    return cvCheckConstraints(cv_mem);
```

Comment:
Shouldn't this call also be wrapped inside `SUNCheckCallLastErrNoRet`?

Reply:
`cvCheckConstraints` does not return a `SUNErrCode`.